### PR TITLE
One hot op

### PR DESCRIPTION
**Pull request by farizrahman4u**
Converts integer indices of shape (batch_size, input_length) to its one hot representation of shape (batch_size, input_length, nb_classes).

### CODE REVIEW

File: keras/backend/theano_backend.py [hunk 1]
```diff
@@ -594,6 +594,15 @@ def spatial_3d_padding(x, padding=(1, 1, 1), dim_ordering='th'):
 def pack(x):
     return T.stack(*x)
 
+
+def one_hot(indices, nb_classes):
+    '''
+    Input: Integer tensor of shape (batch_size, input_length)
+    Output: One hot representation of the input with shape (batch_size, input_length, nb_classes)
+    '''
+    return T.eye(nb_classes, nb_classes)[indices]
```
Comment (fchollet) [inline keras/backend/theano_backend.py]: You should use http://deeplearning.net/software/theano/library/tensor/extra_ops.html#theano.tensor.extra_ops.to_one_hot

**Comment by farizrahman4u** [comment]
@fchollet Done

**Comment by farizrahman4u** [comment]
Hold this PR, do not merge.

**Comment by farizrahman4u** [comment]
`T.extra_ops._to_one_hot` is for a single batch.

**Comment by farizrahman4u** [comment]
@fchollet  All done.

**Comment by fchollet** [comment]
Also, any new backend operation should have a test in `test_backends.py`.

**Comment by farizrahman4u** [comment]
@fchollet What do you call the inverse of one hot function? Is it useful? I have it.

**Comment by fchollet** [comment]
I believe it's called `argmax`. We have it already.

**Comment by farizrahman4u** [comment]
@fchollet Its all green.

**Comment by fchollet** [comment]
I checked it for correctness just now, and it doesn't check out. This was written for one-hot encoding 2D integer tensors into 3D float tensors, which isn't even the most common use case of one-hot encoding (by far).

The most common use case is to one-hot 1D integer tensors (typically labels) into a 2D float tensor. Of course a Keras `one_hot` op should be able to generally one-hot encode `nD` integer tensors into `(n+1)D` float tensors, much like `tf.one_hot`.

- on the TF front, it's already the case, but the docstring is incorrect.
- on the TH front, it isn't the case, because the (niche) 3D use case is hard-coded.
- anyway, we should not have ops that have inconsistent behavior between backends.

**Comment by farizrahman4u** [comment]
@fchollet What about now?

**Comment by farizrahman4u** [comment]
ping @fchollet

**Comment by fchollet** [comment]
LGTM, thanks

**Pull request merged**